source/core/src/main/com/csse3200/game/entities/configs/WeaponConfig.java

```diff
@@ -0,0 +1,153 @@
+        animator.addAnimation("shootDown", 0.05f, Animation.PlayMode.NORMAL);
+        animator.addAnimation("shootLeft", 0.05f, Animation.PlayMode.NORMAL);
+        animator.addAnimation("shootRight", 0.05f, Animation.PlayMode.NORMAL);
+
+        Entity meleeEntity = new Entity()
+                .addComponent(new CollectibleComponent(collectible))
+                .addComponent(new HitboxComponent().setLayer(PhysicsLayer.ITEM))
+                .addComponent(new PhysicsComponent())
+                .addComponent(new ColliderComponent())
+                .addComponent(animator)
+                .addComponent(new WeaponAnimationController());
+
+        PhysicsUtils.setScaledCollider(meleeEntity, -1f, -1f);
+        meleeEntity.getComponent(ColliderComponent.class).setSensor(true);
+        meleeEntity.getComponent(WeaponAnimationRenderComponent.class).startAnimation("idle");
+        meleeEntity.getComponent(ColliderComponent.class).setAsBox(new Vector2(1f, 1f));
+        meleeEntity.getComponent(PhysicsComponent.class).setBodyType(BodyDef.BodyType.StaticBody);
+
+        logger.info("Created melee weapon entity: " + collectible);
+        return meleeEntity;
+    }
+
+    public static Entity createRangeEntity(RangedWeapon collectible) {
+        WeaponAnimationRenderComponent animator =
+                new WeaponAnimationRenderComponent(new TextureAtlas("images/Weapons/plasma_blaster.atlas"));
+        animator.addAnimation("idle", 0.1f, Animation.PlayMode.LOOP);
+        animator.addAnimation("left", 0.04f, Animation.PlayMode.LOOP);
+        animator.addAnimation("up", 0.04f, Animation.PlayMode.LOOP);
+        animator.addAnimation("down", 0.04f, Animation.PlayMode.LOOP);
+        animator.addAnimation("shootUp", 0.05f, Animation.PlayMode.NORMAL);
+        animator.addAnimation("shootDown", 0.05f, Animation.PlayMode.NORMAL);
+        animator.addAnimation("shootLeft", 0.05f, Animation.PlayMode.NORMAL);
+        animator.addAnimation("shootRight", 0.05f, Animation.PlayMode.NORMAL);
+
+        Entity rangedEntity = new Entity()
+                .addComponent(new CollectibleComponent(collectible))
+                .addComponent(new HitboxComponent().setLayer(PhysicsLayer.WEAPON))
+                .addComponent(new PhysicsComponent())
+                .addComponent(new ColliderComponent())
+                .addComponent(animator)
+                .addComponent(new WeaponAnimationController());
+
+        PhysicsUtils.setScaledCollider(rangedEntity, -1f, -1f);
+        rangedEntity.getComponent(ColliderComponent.class).setSensor(true);
+        rangedEntity.getComponent(WeaponAnimationRenderComponent.class).startAnimation("idle");
+        rangedEntity.getComponent(HitboxComponent.class).setSize(new Vector2(3f, 3f));
+        logger.info("Created range weapon entity: " + collectible);
+        return rangedEntity;
+    }
+
+    // Define the WeaponData class
+    public static class WeaponData {
```

Comment:
I don't think this class is needed since we already have the MeleeWeapon and RangedWeapon class, which perform the same functions and are already used in the weaponFactory.